Preprocessing/oxygenPP.m

```diff
@@ -0,0 +1,248 @@
+% ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+% POSSIBILITY OF SUCH DAMAGE.
+%
+narginchk(2, 3);
+
+if ~iscell(sample_data), error('sample_data must be a cell array'); end
+if isempty(sample_data), return;                                    end
+
+% no modification of data is performed on the raw FV00 dataset except
+% local time to UTC conversion
+if strcmpi(qcLevel, 'raw'), return; end
+
+% auto logical in input to enable running under batch processing
+if nargin<3, auto=false; end
+
+for k = 1:length(sample_data)
+  
+  sam = sample_data{k};
+  
+  % Get the temperature
+  tempIdx       = getVar(sam.variables, 'TEMP');
+  if ~(tempIdx), continue; end % if there is no temperature, then don't do anythin  
+  % get coorinates and dimensions from TEMP to use as output
+  % coordinates/dimensions
+  dimensions = sam.variables{tempIdx}.dimensions;
+  if isfield(sam.variables{tempIdx}, 'coordinates')
+      coordinates = sam.variables{tempIdx}.coordinates;
+  else
+      coordinates = '';
+  end
+  temp = sam.variables{tempIdx}.data;
+  
+  % indexs of other variables that might be needed
+  psalIdx       = getVar(sam.variables, 'PSAL');
+  presRelIdx    = getVar(sam.variables, 'PRES_REL');
+  presIdx       = getVar(sam.variables, 'PRES');
+  
+  if (presRelIdx)
+      pres = sam.variables{presRelIdx}.data;
+  else
+      % Should be pressure, convert from depth?
+      pres = sam.instrument_nominal_depth*ones(size(temp));
+  end
+  
+  % oxygen parameters that maybe in the dataset
+  doxIdx        = getVar(sam.variables, 'DOX');
+  dox1Idx       = getVar(sam.variables, 'DOX1');
+  dox2Idx       = getVar(sam.variables, 'DOX2');
+  doxsIdx       = getVar(sam.variables, 'DOXS');
+  oxsolIdx      = getVar(sam.variables, 'OXSOL');
+  
+  % check is psal and some dox measurement present, if not skip
+  if ~(psalIdx && (doxIdx || dox1Idx || dox2Idx || doxsIdx)), continue; end
+  
+%   % get a latitude for calculation of Salinity Absolute from TEOS-10
+%   if isempty(sam.geospatial_lat_min)
+%       disp(['Warning : no geospatial_lat_min, geospatial_long_min documented for oxygenPP to be applied on ' sam.toolbox_input_file]);
+%       prompt = {'Latitude (South -ve)', 'Longitude (West -ve)'};
+%       dlg_title = 'Location';
+%       num_lines = 1;
+%       defaultans = {'0' , '0'};
+%       answer = inputdlg(prompt,dlg_title,num_lines,defaultans);
+%       
+%       % don't try to apply any correction if canceled by user
+%       if isempty(answer), return; end
+%       
+%       sam.geospatial_lat_min = str2double(answer(2));
+%       continue;
+%   end
+%   lat = sam.geospatial_lat_min;
+%   long = sam.geospatial_long_min;
+  
+  psal = sam.variables{psalIdx}.data;
+  
+  ptmp = sw_ptmp(psal, temp , pres, 0);
+  
+  %SA = gsw_SA_from_SP(psal, pres, long, lat);
+  %pt = gsw_pt_from_t(SA, temp, pres, 0);
+  
+  % calculate the density at 0 dbar
+  dens = sw_dens0(psal, ptmp);
```

Comment:
This is from http://archimer.ifremer.fr/doc/00287/39795/ (what argo is now doing).

> Unit conversion of oxygen
> The unit of DOXY is umol/kg in Argo data and the oxygen measurements are sent from Argo floats in another unit such as umol/L for the Optode and ml/L for the SBE-IDO. Thus the unit conversion is carried out by DACs as follows:
> O2 [umol/L] = 44.6596×O2 [ml/L]
> O2 [umol/kg] = O2 [umol/L] / ρ
> Here, ρ is the potential density of water [kg/L] at zero pressure and at the potential temperature (e.g., 1.0269 kg/L; e.g., UNESCO, 1983). The value of 44.6596 is derived from the molar volume of the oxygen gas, 22.3916 L/mole, at standard temperature and pressure (0°C, 1 atmosphere; e.g., García and Gordon, 1992). 
> This unit conversion follows the "Recommendations on the conversion between oxygen quantities for Bio-Argo floats and other autonomous sensor platforms" by SCOR Working Group 142 on "Quality Control Procedures for Oxygen and Other Biogeochemical Sensors on Floats and Gliders" [RD13] . 
> The unit conversion should always be done with the best available temperature, i.e., TEMP of the CTD unit.

The argument to sw_dens0 is practical salinity, is the practical salinity not conservative anyway.